src/App.tsx

```diff
@@ -1,204 +1,112 @@
+    setFilter(selected);
+  };
 
-          {/* This todo is in loadind state */}
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">
-              Todo is being saved now
-            </span>
+  const handleClearCompleted = () => {
+    const updatedTodos = todos.filter(todo => !todo.completed);
 
-            <button type="button" className="todo__remove" data-cy="TodoDelete">
-              ×
-            </button>
+    setTodos(updatedTodos);
+  };
 
-            {/* 'is-active' class puts this modal on top of the todo */}
-            <div data-cy="TodoLoader" className="modal overlay is-active">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-        </section>
-
-        {/* Hide the footer if there are no todos */}
-        <footer className="todoapp__footer" data-cy="Footer">
-          <span className="todo-count" data-cy="TodosCounter">
-            3 items left
-          </span>
-
-          {/* Active link should have the 'selected' class */}
-          <nav className="filter" data-cy="Filter">
-            <a
-              href="#/"
-              className="filter__link selected"
-              data-cy="FilterLinkAll"
-            >
-              All
-            </a>
+  const remainingTodoCount = todos.filter(todo => !todo.completed).length;
 
-            <a
-              href="#/active"
-              className="filter__link"
-              data-cy="FilterLinkActive"
-            >
-              Active
-            </a>
+  return (
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
 
-            <a
-              href="#/completed"
-              className="filter__link"
-              data-cy="FilterLinkCompleted"
-            >
-              Completed
-            </a>
-          </nav>
-
-          {/* this button should be disabled if there are no completed todos */}
-          <button
-            type="button"
-            className="todoapp__clear-completed"
-            data-cy="ClearCompletedButton"
+      <div className="todoapp__content">
+        <NewTodo onAddTodo={handleAddTodo} />
+
+        {todos.length > 0 && (
+          <Filter selectedFilter={filter} onSelectFilter={handleSetFilter} />
+        )}
+
+        {todos.length > 0 && (
+          <TodoList
+            todos={todos}
+            filter={filter}
+            onDeleteTodo={handleDeleteTodo}
+          />
+        )}
+
+        {error && (
+          <div
+            data-cy="ErrorNotification"
+            className="notification is-danger is-light has-text-weight-normal"
           >
```

Comment:
error should looks differently, automatically hide after 3 seconds and you also need error "Title should not be empty"
![image](https://github.com/mate-academy/react_todo-app-loading-todos/assets/147381375/c69a191e-85a7-47ea-a9de-8388222f0551)